src/coreclr/tools/ILVerification/AccessVerificationHelpers.cs

```diff
@@ -22,19 +22,8 @@ internal static bool CanAccess(this TypeDesc currentClass, TypeDesc targetClass)
             if (targetClass.IsParameterizedType)
                 return currentClass.CanAccess(((ParameterizedType)targetClass).ParameterType);
 
-#if false
-            // perform transparency check on the type, if the caller is transparent
-            if ((NULL != pCurrentMD) && Security::IsTransparentMethod(pCurrentMD))
-            {
-                // check if type is visible outside the assembly
-                if (!IsTypeVisibleOutsideAssembly(pTargetClass))
-                {
-                    // check transparent/critical on type
-                    if (!Security::CheckNonPublicCriticalAccess(pCurrentMD, NULL, NULL, pTargetClass)) 
-                        return FALSE;
-                }
-            }
-#endif
+            if (targetClass.IsFunctionPointer)
+                return currentClass.CanAccessSignature(((FunctionPointerType)targetClass).Signature);
```

Comment:
It would be nice to also add a test that exercises `CanAccessSignature` returning false for function pointer here.

Reply:
The negative test in IL is `Dispatcher_Invalid_MethodAccess` which covers the access issue. Do you mean other invalid cases we should add?

Reply:
The function pointer signature in `Dispatcher_Invalid_MethodAccess` is `void *(int32)`. This signature won't fail in function pointer signature access check. `currentClass.CanAccessSignature(((FunctionPointerType)targetClass).Signature)` will return true.

`Dispatcher_Invalid_MethodAccess` fails in owning type access check. It means that it is not actually testing the function pointer access checks.

I meant to add a test that actually fails the function pointer signature access check, ie where this `CanAccessSignature` call returns false.